custom_components/wyzeapi/button.py

```diff
@@ -0,0 +1,145 @@
+        return f"{self._irrigation.nickname} - {self._zone.name}"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID for the zone."""
+        return f"{self._irrigation.mac}-zone-{self._zone.zone_number}"
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this entity."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._irrigation.mac)},
+            name=self._irrigation.nickname,
+            manufacturer="WyzeLabs",
+            model=self._irrigation.product_model,
+            connections={(dr.CONNECTION_NETWORK_MAC, self._device.mac)},
+        )
+
+    @property
+    def device_class(self) -> str:
+        """Return the device class of the button."""
+        return ButtonDeviceClass.RESTART
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return entity specific state attributes."""
+        return {
+            "zone_number": self._zone.zone_number,
+            "zone_id": self._zone.zone_id,
+            "enabled": self._zone.enabled,
+            "quickrun_duration": self._zone.quickrun_duration,
+        }
+
+    async def async_press(self) -> None:
+        """Start the zone with its quickrun duration.
+        
+        This method is called when the button is pressed in Home Assistant.
+        It will start the irrigation zone for the configured quickrun duration.
+        
+        Raises:
+            HomeAssistantError: If the zone cannot be started.
+        """
+        try:
+            await self._irrigation_service.start_zone(
+                self._irrigation,
+                self._zone.zone_number,
+                self._zone.quickrun_duration
+            )
+        except ClientConnectionError as err:
+            raise HomeAssistantError(f"Failed to start zone: {err}") from err
+        except Exception as err:
+            _LOGGER.error("Error starting zone: %s", err)
+            raise HomeAssistantError(f"Failed to start zone: {err}") from err
+
+    @callback
+    def async_update_callback(self, irrigation: IrrigationDevice) -> None:
+        """Update the button's state."""
+        self._irrigation = irrigation
+        # Find the updated zone
+        for zone in irrigation.zones:
+            if zone.zone_number == self._zone.zone_number:
+                self._zone = zone
+                break
+        self.schedule_update_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to updates."""
+        self._irrigation.callback_function = self.async_update_callback
+        self._irrigation_service.register_updater(self._irrigation, 30)
+        await self._irrigation_service.start_update_manager()
+        return await super().async_added_to_hass()
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Clean up when removed."""
+        self._irrigation_service.unregister_updater(self._irrigation) 
```

Comment:
See my comment about updaters, but this is not needed since a button is stateless.